presto-main/src/main/java/io/prestosql/sql/rewrite/ShowStatsRewrite.java

```diff
@@ -128,26 +128,30 @@ protected Node visitShowStats(ShowStats node, Void context)
         {
             checkState(queryExplainer.isPresent(), "Query explainer must be provided for SHOW STATS SELECT");
 
-            if (node.getRelation() instanceof TableSubquery) {
-                Query query = ((TableSubquery) node.getRelation()).getQuery();
-                QuerySpecification specification = (QuerySpecification) query.getQueryBody();
-
-                Plan plan;
+            Query query = getRelationQuery(node);
```

Comment:
`Extract getQueryRelation method` commit:

- the getQueryRelation -> getRelationQuery
- it's not really "Extract... ", because you're also changing flow... Maybe `Factor out getting query for SHOW STATS`